internal/transactionsubmission/transaction_worker.go

```diff
@@ -497,6 +577,81 @@ func (tw *TransactionWorker) saveResponseXDRIfPresent(ctx context.Context, txJob
 	return nil
 }
 
+// monitorPayment sends a metric about a payment tx to the observer, linking it to a entry in the logs that contains specific metadata about said tx.
+func (tw *TransactionWorker) monitorPayment(ctx context.Context, tx *store.Transaction, metricTag monitor.MetricTag, txMetadata *TxMetadata) {
+	eventID := uuid.New().String()
+	labels := map[string]string{
+		"event_id":        eventID,
+		"event_type":      txMetadata.paymentEventType,
+		"tx_id":           tx.ID,
+		"event_time":      time.Now().String(),
+		"app_version":     tw.version,
+		"git_commit_hash": tw.gitCommitHash,
+	}
```

Comment:
Seeing how this method only depends on the TransactionWorker object for the fields `version`, `commit hash` and `metricsService`, I don't think it's worth keeping it inside the TransactionWorker. The worker already has to deal with a lot of complexity to handle the transactions responses, so anything apart from that logic should be encapsulated in a separated module/object. WDYT?

The same goes for `TxMetadata`.

Reply:
great idea, updated!